crates/runtime/src/table.rs

```diff
@@ -212,6 +212,33 @@ impl Table {
         }
     }
 
+    /// Fill `table[dst..]` with values from `items`
+    ///
+    /// Returns a trap error on out-of-bounds accesses.
+    pub fn init_funcs(
+        &self,
+        dst: u32,
+        items: impl ExactSizeIterator<Item = *mut VMCallerCheckedAnyfunc>,
+    ) -> Result<(), Trap> {
+        let ty = TableElementType::Func;
+        assert!(ty == TableElementType::Func);
+
+        self.with_elements_mut(|elements| {
+            let elements = match elements
+                .get_mut(usize::try_from(dst).unwrap()..)
+                .and_then(|s| s.get_mut(..items.len()))
+            {
+                Some(elements) => elements,
+                None => return Err(Trap::wasm(ir::TrapCode::TableOutOfBounds)),
+            };
+
+            for (item, slot) in items.zip(elements) {
+                Self::set_raw(ty, slot, item.into())
```

Comment:
Does this get boiled down into a `memcpy` by LLVM? If so, we can close #983. If not, would we be able to do that here?

Reply:
It seems like this has just one call site, and I don't think anything is preventing us from doing a `memcpy` in theory here.

Reply:
Oh this should actually probably close that issue. We don't actually have anything to `memcpy` though because our element segments are stored as lists of function indices, but the tables themselves are `*mut VMCallerCheckedAnyfunc`. This means that there's a bit of a mismatch where unless we allocate table segments per-instance to `memcpy` from (which I dont think we should do) then there's actually nothing to memcpy. 

Otherwise, though, I just used `set_raw` here to be the same as the other bits and pieces of this module

Reply:
I went ahead though and made it a bit more raw to help streamline this a tiny bit more